design/core_collection_operations.md

```diff
@@ -0,0 +1,175 @@
+      versions.
+  * The user specifies the version, which to the collections API is an opaque string. Any
+    semantics are supplied by the user.
+    * Note that this means semantic versions will not sort in order - the DB and API would need
+      specific schemas / code to do that.
+* Activate a collection version.
+  * Administration only operation.
+  * Can activate any version at any time.
+  * The active version is the only version accessible to users.
+  * Separating the saving and activation steps means we can save a collection document,
+    load data products for that collection, and then activate the collection when the data
+    products are ready.
+  * It also means we can trivially roll back changes.
+* Get a collection version.
+  * Public operation.
+  * Initially, this is only the active collection.
+    * At some point we may wish to supply admin endpoints for accessing alternate collection
+      versions.
+* List collections
+  * Public operation
+  * Only lists active collections.
+* Longer term:
+  * List all collections, even those without active versions.
+    * Administration only operation.
+    * For now, can be done relatively easily via the Aardvark Arango UI.
+  * List collection versions
+    * Administration only operation.
+    * This may be a long list. Returning only the list of version strings for one collection 
+      is probably reasonable up to ~100K versions, which seems much larger than we'll ever
+      encounter.
+
+## Collection contents
+
+For a minimal start, more will be added later (e.g. data comparison methods, contributors, etc.)
+
+* ID (lowercase alpha only string, no punctuation other than underscores, otherwise opaque)
+* Name (opaque string)
+* Version (opaque string)
+* Source version (opaque string)
+  * The source version may be the same from collection version to collection version if the
+    same source data is reloaded to, say, fix bugs or add features.
+  * This implies a particular collection version should not mix data from different source
+    versions.
+* Icon url
+  * Could probably use FastAPI's built in static server to serve these for now
+* Creation date
+* (If active) activation date
+* List of available data products with the active version of each data product (see example
+  below)
+  * Generally speaking, each data product will support a view in the collections UI
+  * The version information allows creating a new version of the collection document but only
+    updating some of the associated data products, while leaving others at their current
+    version
+  * Similar to the collections version, it also allows for easy rollbacks, and any version
+    semantics are up to the user.
+  * In the future, we could add a schema version if the schema needs to change. For now, no
+    schema version implies a schema version of 1.
+    * An altenative would just be to add a version to the product name, like `taxa_freq2`.
+
+```
+[
+    {
+        "product": "taxa_freq",
```

Comment:
is this the "data_product_id" specified in the endpoint below? if so should we maybe call that path argument "data_product_type"? Or is there going to be an ID field that is unique across all collections that is listed in the `/collection/{collection_id}` endpoint?

Reply:
> is this the "data_product_id" specified in the endpoint below?

Right

> if so should we maybe call that path argument "data_product_type"? 

Hmm, what abour just `data_product`?

Reply:
> Hmm, what abour just data_product?
perfect

Reply:
done